server/src/test/java/org/elasticsearch/tasks/TaskManagerTests.java

```diff
@@ -260,6 +272,95 @@ public void testTaskAccounting() {
         assertNull(taskManager.childTasksPerConnection(task1.getId(), connection1));
     }
 
+    /**
+     * Check that registering a task also causes tracing to be started on that task.
+     */
+    public void testRegisterTaskStartsTracing() {
+        final Tracer mockTracer = Mockito.mock(Tracer.class);
+        final TaskManager taskManager = new TaskManager(Settings.EMPTY, threadPool, Set.of(), mockTracer);
+
+        final Task task = taskManager.register("testType", "testAction", new TaskAwareRequest() {
+
+            @Override
+            public void setParentTask(TaskId taskId) {}
+
+            @Override
+            public TaskId getParentTask() {
+                return TaskId.EMPTY_TASK_ID;
+            }
+        });
+
+        verify(mockTracer).onTraceStarted(any(), eq("task-" + task.getId()), eq("testAction"), anyMap());
+    }
+
+    /**
+     * Check that unregistering a task also causes tracing to be stopped on that task.
+     */
+    public void testUnregisterTaskStopsTracing() {
+        final Tracer mockTracer = Mockito.mock(Tracer.class);
+        final TaskManager taskManager = new TaskManager(Settings.EMPTY, threadPool, Set.of(), mockTracer);
+
+        final Task task = taskManager.register("testType", "testAction", new TaskAwareRequest() {
+
+            @Override
+            public void setParentTask(TaskId taskId) {}
+
+            @Override
+            public TaskId getParentTask() {
+                return TaskId.EMPTY_TASK_ID;
+            }
+        });
+
+        taskManager.unregister(task);
+
+        verify(mockTracer).onTraceStopped("task-" + task.getId());
+    }
+
+    /**
+     * Check that registering and executing a task also causes tracing to be started and stopped on that task.
+     */
+    public void testRegisterAndExecuteStartsAndStopsTracing() {
+        final Tracer mockTracer = Mockito.mock(Tracer.class);
+        final TaskManager taskManager = new TaskManager(Settings.EMPTY, threadPool, Set.of(), mockTracer);
+
+        final Task task = taskManager.registerAndExecute(
+            "testType",
+            new TransportAction<ActionRequest, ActionResponse>("actionName", new ActionFilters(Set.of()), taskManager) {
+                @Override
+                protected void doExecute(Task task, ActionRequest request, ActionListener<ActionResponse> listener) {
+                    listener.onResponse(new ActionResponse() {
+                        @Override
+                        public void writeTo(StreamOutput out) {}
+                    });
+                }
+            },
+            new ActionRequest() {
+                @Override
+                public ActionRequestValidationException validate() {
+                    return null;
+                }
+
+                @Override
+                public TaskId getParentTask() {
+                    return TaskId.EMPTY_TASK_ID;
+                }
+            },
+            null,
+            new ActionListener<>() {
```

Comment:
If you want this shorter, you can use `ActionTestUtils.assertNoFailure(r -> {})`